### PR TITLE
Feature/courses redux

### PR DESCRIPTION
**Pull request by JorgeEvangelista500**
## Description
Created Redux actions and reducers for courses, enable full CRUD operations for courses, hitting the backend. Changes need to be made to the instructor add form, for the data to match the backend

Fixes # (issue)

## Loom Video

https://www.loom.com/share/73be9126ecc949629d70e8a05f4294f3

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/InstructorAddCourse/InstructorAddCourseForm.js [hunk 1]
```diff
@@ -144,7 +153,7 @@ const InstructorAddCourseForm = props => {
                     <label for="classLink">Course Name</label>
                     <Input
                       value={classData.course_name}
-                      name="location"
+                      name="course_name"
```
Comment (lizzythomson) [inline src/components/pages/InstructorAddCourse/InstructorAddCourseForm.js:156]: Good catch!

File: src/components/pages/InstructorAddCourse/InstructorAddCourseForm.js [hunk 2]
```diff
@@ -376,7 +385,7 @@ const InstructorAddCourseForm = props => {
               type="submit"
               htmlType="submit"
               className="submitButton"
-              onClick={success}
```
Comment (lizzythomson) [inline src/components/pages/InstructorAddCourse/InstructorAddCourseForm.js:379]: Are we no longer using the success function on line 108? If so, should delete it?

File: src/redux/actions/coursesActions.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  try {
+    const res = await axiosWithAuth(idToken).get(`/courses`);
+    dispatch({
+      type: GET_COURSES,
+      payload: res.data,
+    });
+  } catch (error) {
+    dispatch({
+      payload: error.message,
+    });
```
Comment (lizzythomson) [inline src/redux/actions/coursesActions.js:28]: It looks like this code is dispatching an action without a type, so it isn't doing anything. Do we need to add an action type for errors and a case in the switch inside the reducer?

**Review by lizzythomson — CHANGES_REQUESTED** [review]
Other than a couple comments everything looks great!

**Review by Berenika14 — APPROVED** [review]
You did an amazing job by building the feature of courses using redux adding functionality to this specific feature. Only make sure you look at the changes that Lizzy requested, overall great work.

File: src/components/pages/InstructorAddCourse/InstructorAddCourseForm.js (hunk 2 above)
Comment (JorgeEvangelista500) [reply src/components/pages/InstructorAddCourse/InstructorAddCourseForm.js:379]: thank you I missed that, moved the success modal into the submit function

**Review by lizzythomson — APPROVED** [review]

**Pull request merged**